src/query/storages/fuse/src/operations/mutation/compact/compact_part.rs

```diff
@@ -13,23 +13,67 @@
 // limitations under the License.
 
 use std::any::Any;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::Hash;
+use std::hash::Hasher;
 use std::sync::Arc;
 
 use common_catalog::plan::PartInfo;
 use common_catalog::plan::PartInfoPtr;
 use common_exception::ErrorCode;
 use common_exception::Result;
 use storages_common_table_meta::meta::BlockMeta;
+use storages_common_table_meta::meta::CompactSegmentInfo;
+use storages_common_table_meta::meta::Statistics;
 
 use crate::operations::common::BlockMetaIndex;
+use crate::operations::mutation::BlockIndex;
+use crate::operations::mutation::SegmentIndex;
+
+#[derive(serde::Serialize, serde::Deserialize, PartialEq, Clone)]
+pub struct CompactLazyPartInfo {
+    pub segment_indices: Vec<SegmentIndex>,
+    pub compact_segments: Vec<Arc<CompactSegmentInfo>>,
+}
+
+#[typetag::serde(name = "compact_lazy")]
+impl PartInfo for CompactLazyPartInfo {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn equals(&self, info: &Box<dyn PartInfo>) -> bool {
+        info.as_any()
+            .downcast_ref::<CompactLazyPartInfo>()
+            .is_some_and(|other| self == other)
+    }
+
+    fn hash(&self) -> u64 {
+        let mut s = DefaultHasher::new();
+        self.segment_indices.hash(&mut s);
+        s.finish()
+    }
+}
+
+impl CompactLazyPartInfo {
+    pub fn create(
+        segment_indices: Vec<SegmentIndex>,
+        compact_segments: Vec<Arc<CompactSegmentInfo>>,
+    ) -> PartInfoPtr {
+        Arc::new(Box::new(CompactLazyPartInfo {
+            segment_indices,
+            compact_segments,
+        }))
+    }
+}
 
 #[derive(serde::Serialize, serde::Deserialize, PartialEq)]
-pub struct CompactPartInfo {
-    pub blocks: Vec<Arc<BlockMeta>>,
-    pub index: BlockMetaIndex,
+pub enum CompactPartInfo {
+    CompactExtraInfo(CompactExtraInfo),
+    CompactTaskInfo(CompactTaskInfo),
 }
```

Comment:
what's the difference of CompactExtraInfo and CompactTaskInfo?

Reply:
> what's the difference of CompactExtraInfo and CompactTaskInfo?

CompactExtraInfo: the extra info for compact, such as unchanged blocks, removed segment indices, statistics.
CompactTaskInfo: the compact task